src/Microsoft.ML.Core/Utilities/ThreadUtils.cs

```diff
@@ -4,44 +4,154 @@
 
 using System;
 using System.Collections.Generic;
-using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.ML.Internal.Utilities
 {
     internal static partial class Utils
     {
-        public static Thread CreateBackgroundThread(ParameterizedThreadStart start)
-        {
-            return new Thread(start)
-            {
-                IsBackground = true
-            };
-        }
+        public static Task StartBackgroundThread(Action start) =>
```

Comment:
>StartBackgroundThread [](start = 27, length = 21)

Is the fact that this thing is returning a task, yet the name of the thing still has thread it in, potentially confusing? This is not part of our public API, but might cause some confusion.

Reply:
Ok, I'll rename this to RunOnBackgroundThread or something like that.